core/domain/collection_domain.py

```diff
@@ -563,7 +563,8 @@ def from_dict(
             collection_dict['schema_version'],
             [
                 CollectionNode.from_dict(node_dict)
-                for node_dict in collection_dict['nodes']
+                for node_dict in (
```

Comment:
I'm not entirely sure why we're doing this calculation here. Shouldn't we only do it just before we pass the collection to the frontend?